mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/StatefulSettingsScreenTest.kt

```diff
@@ -102,4 +101,31 @@ class StatefulSettingsScreenTest {
 
     Truth.assertThat(functionCalled).isTrue()
   }
+
+  @Test
+  fun given_statefulSettingsScreen_when_logoutButtonClicked_then_redirectToAuthenticationScreen() =
+      runTest {
+    val auth = buildAuth { user("email@example.org", "password", "1") }
+    val store = emptyStore()
+    val authFacade = AuthenticationFacade(auth, store)
+    authFacade.signInWithEmail("email@example.org", "password")
+
+    val user = authFacade.currentUser.filterIsInstance<AuthenticatedUser>().first()
+
+    val userMock = mockk<AuthenticatedUser>()
+    every { userMock.name } returns "test"
+    every { userMock.email } returns "test"
+    every { userMock.emoji } returns "test"
+    every { userMock.backgroundColor } returns Profile.Color.Orange
+    every { userMock.uid } returns user.uid
+    every { userMock.followed } returns false
+    coEvery { userMock.signOut() } coAnswers { user.signOut() }
+
+    val testEnv =
+        rule.setContentWithTestEnvironment(auth = auth) {
+          StatefulSettingsScreen(userMock, {}, {}, {})
+        }
+    rule.onNodeWithText(testEnv.strings.settingLogout).assertExists().performClick()
+    coVerify { userMock.signOut() }
+  }
```

Comment:
I feel like this test could be simplified to something like

```kotlin
  @Test
  fun given_statefulSettingsScreen_when_logoutButtonClicked_then_disconnectsUser() = runTest {
    val env = rule.setContentWithTestEnvironment { StatefulSettingsScreen(user, {}, {}, {}) }
    rule.onNodeWithText(env.strings.settingLogout).assertExists().performClick()
    assertThat(env.facades.auth.currentUser.first()).isEqualTo(NotAuthenticatedUser)
  }
```.

Wdyt ?

Reply:
Yes indeed much cleaner thank you !